### PR TITLE
[DRAFT] Feat/mynw 111 private shard support

### PR DESCRIPTION
**Pull request by followdarko**
Context:
https://docs.google.com/document/d/1iU2uu_63wFqwWuypCCcpLxeKZtSy8gxpvOQXS8ZPvuk/edit
https://www.notion.so/kikimora-labs/Private-Shards-Support-43663afa7bfb41bba8afb21a92857a16
Jira https://mnw.atlassian.net/browse/MYNW-111

TODO:

- [x] Should we do anything with icon and colours of shard in UI?
- [x] Tooltip text for private shard info in UI
- [x] Should private shard RPC url be a link?
- [ ] Should we add additional check where account id ends with private shard id?
- [x] If account that not belongs to private shard try to sign a transaction to it what should we display?
- [x] We should add analytics to sign transactions on private shards. @PinkiNice 

Need feedback @PinkiNice @xilosada

### CODE REVIEW

File: packages/frontend/src/utils/wallet.js [hunk 1]
```diff
@@ -1121,6 +1121,30 @@ export default class Wallet {
...
 
+    // TODO: just a copy-paste draft
+    async signAndSendCalimeroTransaction(transactions, accountId = this.accountId, customRPCUrl) {
+        // TODO: no need to store hashes since it's private shard
+        const transactionHashes = [];
+        const calimeroConnection = nearApiJs.providers.JsonRpcProvider(customRPCUrl);
+        for (let { receiverId, nonce, blockHash, actions } of transactions) {
+            let status, transaction;
+
+            const [, signedTransaction] = await nearApiJs.transactions.signTransaction(receiverId, nonce, actions, blockHash, calimeroConnection.signer, accountId, NETWORK_ID);
```
Comment (xilosada) [inline packages/frontend/src/utils/wallet.js:1132]: signer not defined

**Review by xilosada — CHANGES_REQUESTED** [review]
This is great I used it as base and opened a PR with the changes required to make it work e2e

File: packages/frontend/src/utils/wallet.js [hunk 2]
```diff
@@ -1121,6 +1121,30 @@ export default class Wallet {
         return transactionHashes;
     }
 
+    // TODO: just a copy-paste draft
+    async signAndSendCalimeroTransaction(transactions, accountId = this.accountId, customRPCUrl) {
+        // TODO: no need to store hashes since it's private shard
+        const transactionHashes = [];
+        const calimeroConnection = nearApiJs.providers.JsonRpcProvider(customRPCUrl);
```
[the remark this answers is not in the document]
Comment (followdarko) [reply packages/frontend/src/utils/wallet.js:1128]: @xilosada I'm sorry where did you replaced it?

[the remark this answers is not in the document]
Comment (followdarko) [reply packages/frontend/src/utils/wallet.js]: I updated to `nearApiJs.Connection.fromConfig`
Not sure about `x-api-token` header thought

**Comment by xilosada** [comment]
@followdarko We will address all these questions after Nearcon.

**Comment by PinkiNice** [comment]
- Should we do anything with icon and colours of shard in UI?
Out of scope for now
 - Tooltip text for private shard info in UI
 "Private shard transactions are executed inside private shard environment. They will not affect you main account."
- Should private shard RPC url be a link?
No, lets make it just a text
- Should we add additional check where account id ends with private shard id?
We should check "signer" and "receiver" in transaction to be under "private-shard-id" scope
- If account that not belongs to private shard try to sign a transaction to it what should we display?
Signing account always will be .testnet or .mainnet  (not private shard). So it is ok
- We should add analytics to sign transactions on private shards.
Out of scope for now

**Comment by followdarko** [comment]
@xilosada is it ok display full path rpc url or should we make it shorter?
Screenshot 2022-09-08 at 23 41 09

**Pull request merged**